### PR TITLE
Optimize deg2rad/rad2deg dispatch

### PR DESCRIPTION
**Pull request by kumargope**
### Summary
Replaces the expensive `BinaryNgDeviceOperation` dispatch path in `ttnn.deg2rad` and `ttnn.rad2deg` with native unary scalar multiplication (`MUL_UNARY_SFPU`).

### Problem
`ttnn.deg2rad` and `ttnn.rad2deg` previously dispatched as binary multiplication operations with a scalar parameter. This instantiated a binary operation path (`BinaryNgDeviceOperation`), incurring unnecessary dispatch overhead.

### Solution
- Updated `ttnn::deg2rad` and `ttnn::rad2deg` in C++ to use `UnaryOpType::MUL_UNARY_SFPU` via `operations::unary::detail::unary_impl`.
- Updated `deg2rad_bw` and `rad2deg_bw` in `unary_backward.cpp` to call `ttnn::deg2rad` and `ttnn::rad2deg` directly.
- Added unit tests in `test_unary.py` verifying numerical accuracy across float32/bfloat16 dtypes, representative shapes, and key angle/radian values.

### Testing
Added `test_angle_conversion_values_and_dtypes` in `tests/ttnn/unit_tests/operations/eltwise/test_unary.py` asserting PCC >= 0.99 against PyTorch golden references.

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py [hunk 1]
```diff
@@ -2694,3 +2694,49 @@ def test_softcap_zero_beta_guard(device, expect_error):
...
+                -3.141592653589793 / 2,
+                -3.141592653589793,
+                100.0,
+                -100.0,
+            ],
+            ttnn.rad2deg,
+        ),
+    ],
+)
+@pytest.mark.parametrize("shape", [[1, 1, 32, 32], [1, 1, 256, 256], [1, 4, 512, 512]])
```
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary.py]: Pls remove this shape [1, 4, 512, 512]

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py [hunk 2]
```diff
@@ -2694,3 +2694,49 @@ def test_softcap_zero_beta_guard(device, expect_error):
...
+@pytest.mark.parametrize("torch_dtype, ttnn_dtype", [(torch.float32, ttnn.float32), (torch.bfloat16, ttnn.bfloat16)])
+def test_angle_conversion_values_and_dtypes(
+    device, op_name, values, ttnn_op, shape, torch_dtype, ttnn_dtype
+):
+    num_elements = 1
+    for dim in shape:
+        num_elements *= dim
+
+    repeated_values = (values * (num_elements // len(values) + 1))[:num_elements]
+    torch_input_tensor = torch.tensor(repeated_values, dtype=torch_dtype).reshape(shape)
```
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary.py]: instead of single values, 
Could you use generate_bfloat16_bits_in_range

**Review by KalaivaniMCW — CHANGES_REQUESTED** [review]
The added unit tests do not test the backward ops that are modified, but their forward counterparts

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py (hunk 1 above)
Comment (kumargope) [reply tests/ttnn/unit_tests/operations/eltwise/test_unary.py]: "Removed [1, 4, 512, 512] shape from the test parametrization as requested."

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py (hunk 2 above)
Comment (kumargope) [reply tests/ttnn/unit_tests/operations/eltwise/test_unary.py]: "Updated bfloat16 input data generation to use generate_bfloat16_bits_in_range."

**Comment by kumargope** [comment]
Added dedicated unit tests `test_angle_conversion_bw_ops` for the modified backward operations `deg2rad_bw` and `rad2deg_bw`. Updated the PR with these changes!

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py [hunk 3]
```diff
@@ -2694,3 +2695,50 @@ def test_softcap_zero_beta_guard(device, expect_error):
...
+        torch_input_tensor = torch.empty(shape, dtype=torch_dtype).uniform_(low, high)
+
+    golden_function = ttnn.get_golden_function(ttnn_op)
+    golden_output = golden_function(torch_input_tensor, device=device)
+
+    input_tensor = ttnn.from_torch(torch_input_tensor, dtype=ttnn_dtype, layout=ttnn.TILE_LAYOUT, device=device)
+    tt_output = ttnn_op(input_tensor)
+    torch_output = ttnn.to_torch(tt_output)
+
+    assert_with_pcc(golden_output, torch_output, pcc=0.99)
```
Comment (KalaivaniMCW) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary.py:2725]: have you tried assert_with_ulp 
what are the ULP errors and max ULP like ?
